frontend/src/router/index.ts

```diff
@@ -129,13 +153,27 @@ const router = createRouter({
 });
 
 router.beforeEach(async (to, from) => {
+    // clear any previous error message
+    clearToastError();
+
     // Refresh token first before navigation.
     if (AuthService.state.value.famLoginUser) {
         // condition needed to prevent infinite redirect
         await AuthService.methods.refreshToken();
+    } else if (to.meta?.layout == 'ProtectedLayout') {
+        // if user tries to access the pages need permission,
+        // will rediret the user back to the landing page, and pop up the error message
+        const { showToastErrorTopRight } = useToastService();
+        showToastErrorTopRight('You are not logged in. Please log in.');
+        router.push('/');
     }
 });
 
+router.afterEach((to, from) => {
+    const { showToastErrorTopRight } = useToastService();
+    if (toastError.value !== '') showToastErrorTopRight(toastError.value);
```

Comment:
after we enter the router, if we set any error message in the beforeEnter method, it will be pop up.